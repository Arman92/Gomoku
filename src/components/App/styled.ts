import styled from "styled-components";

export const AppContainer = styled.div`
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;
