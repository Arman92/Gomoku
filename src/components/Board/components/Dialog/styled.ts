import styled from "styled-components";
import DialogContent from '@material-ui/core/DialogContent';
import Button from '@material-ui/core/Button';

export const StyledDialogContent = styled(DialogContent)`
  margin-top: 2rem;
  text-align: center;
`;

export const StyledButton = styled(Button)`
  &.MuiButton-textSizeLarge {
    font-size: 1.6rem;
  }
`;
