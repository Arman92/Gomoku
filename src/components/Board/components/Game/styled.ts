import styled from "styled-components";
import { Player } from "@gobang/utils/GameBoard/types";

type GameProps = {
  length: number;
};

type ButtonProps = {
  turn: Player;
};

export const GameContainer = styled.div<GameProps>`
  width: 100%;
  height: 100%;
  object-fit: contain;
  display: grid;
  grid-template-rows: repeat(${props => props.length}, minmax(1rem, 1fr));
  grid-template-columns: repeat(${props => props.length}, minmax(1rem, 1fr));
`;

export const CellContainer = styled.div<GameProps>`
  position: relative;
  z-index: 0;

  &:before {
    content: "";
    position: absolute;
    border-bottom: 1px black solid;
    height: 50%;
    width: 100%;
    z-index: -1;
  }

  &:after {
    content: "";
    position: absolute;
    border-right: 1px black solid;
    height: 100%;
    width: 50%;
    z-index: -1;
  }

  /* first row of the grid */
  &:nth-child(-n + ${props => props.length}) {
    &:after {
      height: 50%;
      bottom: 0;
    }
  }

  /* last column of the grid */
  &:nth-child(${props => props.length}n) {
    &:before {
      width: 50%;
    }
  }

  /* first column of the grid */
  &:nth-child(${props => props.length}n + 1) {
    &:before {
      width: 50%;
      left: 50%;
    }
  }

  /* last row of the grid */
  &:nth-child(n + ${props => props.length * (props.length - 1) + 1}) {
    &:after {
      height: 50%;
      top: 0;
    }
  }
`;

export const CellButton = styled.button<ButtonProps>`
  width: 75%;
  height: 75%;
  background: transparent;
  opacity: 0.6;
  border: none;
  position: absolute;
  cursor: pointer;
  transform: translate(15%, 15%);
  border-radius: 100%;
  z-index: 1;

  &.btn-free {
    &:hover {
      background: black;
      background: ${props =>
        props.turn === Player.WHITE
          ? "radial-gradient(circle at 30% 30%, white, #868484)"
          : "radial-gradient(circle at 30% 30%, #5f5f5f, #000)"};
    }

    &:active {
      opacity: 1;
      background: black;
      background: ${props =>
        props.turn === Player.WHITE
          ? "radial-gradient(circle at 30% 30%, white, #868484)"
          : "radial-gradient(circle at 30% 30%, #5f5f5f, #000)"};
    }
  }

  &:focus {
    outline: none;
  }

  &.btn-selected-white {
    opacity: 1;
    background: radial-gradient(circle at 30% 30%, white, #868484);
    box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.5);
  }

  &.btn-selected-black {
    opacity: 1;
    background: radial-gradient(circle at 30% 30%, #5f5f5f, #000);
    box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.5);
  }

  &.btn-winner-cell {
    box-shadow: 0 0 5px 3px #ff003b;
  }

  &:disabled:hover {
    cursor: not-allowed;
  }

  @keyframes pulse {
    0% {
      border-color: rgba(18, 255, 235, 0.8);
      background: transparent;
      transform: scale(1.1) rotateZ(360deg);
    }
    to {
      border-color: transparent;
      border-width: 6px;
      transform: scale(1.9) rotateZ(360deg);
    }
  }

  &.btn-last-played:after {
    content: "";
    width: 100%;
    height: 100%;
    display: block;
    border-radius: 100%;
    z-index: -1;
    border: 2px solid rgba(239, 241, 242, 0);
    animation: pulse 0.9s 0.4s infinite ease-in-out;
    backface-visibility: hidden;
  }
`;
