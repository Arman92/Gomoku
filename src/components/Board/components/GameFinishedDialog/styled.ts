import styled from "styled-components";
import DialogContent from "@material-ui/core/DialogContent";

export const StyledDialogContent = styled(DialogContent)`
  margin-top: 2rem;
  text-align: center;
`;
