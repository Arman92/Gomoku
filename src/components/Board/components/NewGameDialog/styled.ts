import styled from "styled-components";

export const FormRow = styled.div`
  margin-top: 3rem;
`;
