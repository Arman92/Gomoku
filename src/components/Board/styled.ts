import styled from "styled-components";
import backgroundImage from "@gobang/assets/imgs/14103.jpg";

type BoardProps = {
  rows: number;
  columns: number;
};

export const BoardContainer = styled.div<BoardProps>`
  display: grid;
  grid-template-rows: repeat(${props => props.rows}, 4.5rem);
  grid-template-columns: repeat(${(props: any) => props.columns}, 4.5rem);
  /* background-image: url(${backgroundImage}); */
  background: #e6e6e6;
  /* grid-gap: 1rem; */
  padding: 10rem;

`;

export const ButtonContainer = styled.div`
  position: relative;
  z-index: 0;

  &:before {
    content: "";
    position: absolute;
    border-bottom: 1px black solid;
    height: 50%;
    width: 100%;
    z-index: -1;
  }

  &:after {
    content: "";
    position: absolute;
    border-right: 1px black solid;
    height: 100%;
    width: 50%;
    z-index: -1;
  }

  /* first row of the grid */
  &:nth-child(-n + 8) {
    &:after {
      height: 50%;
      bottom: 0;
    }
  }

  /* last column of the grid */
  &:nth-child(8n) {
    &:before {
      width: 50%;
    }
  }

  /* first column of the grid */
  &:nth-child(8n + 1) {
    &:before {
      width: 50%;
      left: 50%;
    }
  }

  /* last row of the grid */
  &:nth-child(n + 57) {
    &:after {
      height: 50%;
      top: 0;
    }
  }
`;

export const Button = styled.button`
  width: 95%;
  height: 95%;
  background: gray;
  opacity: 0.6;
  border: none;
  position: absolute;
  top: 0;
  left: 0;
  cursor: pointer;
  transform: translate(2.5%, 2.5%);
  border-radius: 95%;
  z-index: 1;
`;
