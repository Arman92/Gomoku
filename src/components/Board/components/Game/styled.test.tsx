import React from "react";
import "jest-styled-components";

import { mount } from "enzyme";

import { GameContainer, CellContainer } from "./styled";

describe("Game Styled Components", () => {
  it("It should render GameContainer correctly", () => {
    const length = 10;
    const gameContainerWrapper = mount(<GameContainer length={length} />);
    expect(gameContainerWrapper.find(GameContainer)).toMatchSnapshot();

    expect(gameContainerWrapper).toHaveStyleRule(
      "grid-template-rows",
      `repeat(${length},minmax(1rem,1fr))`
    );

    expect(gameContainerWrapper).toHaveStyleRule(
      "grid-template-columns",
      `repeat(${length},minmax(1rem,1fr))`
    );
  });

  it("It should render CellContainer correctly", () => {
    const length = 10;
    const cellContainerWrapper = mount(<CellContainer length={length} />);
    expect(cellContainerWrapper.find(CellContainer)).toMatchSnapshot();
  });
});
