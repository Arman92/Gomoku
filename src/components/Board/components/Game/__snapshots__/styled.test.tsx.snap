// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Styled Components It should render CellContainer correctly 1`] = `
.c0 {
  position: relative;
  z-index: 0;
}

.c0:before {
  content: "";
  position: absolute;
  border-bottom: 1px black solid;
  height: 50%;
  width: 100%;
  z-index: -1;
}

.c0:after {
  content: "";
  position: absolute;
  border-right: 1px black solid;
  height: 100%;
  width: 50%;
  z-index: -1;
}

.c0:nth-child(-n + 10):after {
  height: 50%;
  bottom: 0;
}

.c0:nth-child(10n):before {
  width: 50%;
}

.c0:nth-child(10n + 1):before {
  width: 50%;
  left: 50%;
}

.c0:nth-child(n + 91):after {
  height: 50%;
  top: 0;
}

<styled.div
  length={10}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  position: relative;
  z-index: 0;

  &:before {
    content: \\"\\";
    position: absolute;
    border-bottom: 1px black solid;
    height: 50%;
    width: 100%;
    z-index: -1;
  }

  &:after {
    content: \\"\\";
    position: absolute;
    border-right: 1px black solid;
    height: 100%;
    width: 50%;
    z-index: -1;
  }

  /* first row of the grid */
  &:nth-child(-n + ",
            [Function],
            ") {
    &:after {
      height: 50%;
      bottom: 0;
    }
  }

  /* last column of the grid */
  &:nth-child(",
            [Function],
            "n) {
    &:before {
      width: 50%;
    }
  }

  /* first column of the grid */
  &:nth-child(",
            [Function],
            "n + 1) {
    &:before {
      width: 50%;
      left: 50%;
    }
  }

  /* last row of the grid */
  &:nth-child(n + ",
            [Function],
            ") {
    &:after {
      height: 50%;
      top: 0;
    }
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    length={10}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</styled.div>
`;

exports[`Game Styled Components It should render GameContainer correctly 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  object-fit: contain;
  display: grid;
  grid-template-rows: repeat(10,minmax(1rem,1fr));
  grid-template-columns: repeat(10,minmax(1rem,1fr));
}

<styled.div
  length={10}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  width: 100%;
  height: 100%;
  object-fit: contain;
  display: grid;
  grid-template-rows: repeat(",
            [Function],
            ", minmax(1rem, 1fr));
  grid-template-columns: repeat(",
            [Function],
            ", minmax(1rem, 1fr));
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    length={10}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</styled.div>
`;
